src/txdb.cpp

```diff
@@ -248,6 +249,9 @@ bool CBlockTreeDB::ReadFlag(const std::string &name, bool &fValue) {
 
 bool CBlockTreeDB::LoadBlockIndexGuts(const Consensus::Params& consensusParams, std::function<CBlockIndex*(const uint256&)> insertBlockIndex)
 {
+    std::string thread_name = util::ThreadGetInternalName();
+    util::ThreadRename("leveldb");
```

Comment:
Why is this needed? Also, calling this leveldb is a layer-violation.

Reply:
leveldb spawn a thread which inherits current thread name: "bitcoind" or "qt-init". Therefore, on master the OS reports about two "bitcoind" threads.

Reply:
Oh, I see. Maybe add a comment?

Reply:
Mind suggesting a good wording?

Reply:
To be honest, I think it's better to leave this out than do an ugly layer violation like this.
At the least this would need to be documented very well, just having a scattered thread rename is confusing for anyone reading the code in the future.
But I wonder if there's a better way to hook leveldb's thread creation and rename the threads.

Reply:
Is it ok to just drop a commit with this change for better solution?

Reply:
It seemed weird anyway to to gui changes in the same pull request with leveldb changes

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18790#issuecomment-622099051).